plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -3846,4 +3892,33 @@ public boolean supportsReportingWrittenBytes(ConnectorSession session, SchemaTab
     {
         return true;
     }
+
+    public static class TableParameterProvider
+    {
+        private final Table table;
+        private final Set<String> consumedParameters = new HashSet<>(TABLE_PROPERTIES_TO_SKIP);
+
+        public TableParameterProvider(Table table)
+        {
+            this.table = requireNonNull(table, "table is null");
+        }
+
+        public String getParameter(String key)
+        {
+            consumedParameters.add(key);
+            return table.getParameters().get(key);
+        }
+
+        public Table getTable()
+        {
+            return table;
+        }
+
+        public Map<String, String> getUnconsumedProperties()
+        {
+            return table.getParameters().entrySet().stream()
+                    .filter(entry -> !consumedParameters.contains(entry.getKey()))
```

Comment:
Should we lowercase the values entered in and also the values used for checking  here just to be on the safe side?